[MIG] l10n_ar_sale_order_type: Migration to 18.0

mem-adhoc opened the pull request.

Commit:
[MIG] l10n_ar_sale_order_type: Migration to 18.0

roboadhoc commented:
[Pull request status dashboard](https://mergebot.adhoc.ar/ingadhoc/argentina-sale/pull/217).

jjscarafia committed:
[REM] l10n_ar_sale: deprecate sale_checkbook

Sale checkbook servía principalmente para dar una numeración distinta. Este feature ya lo tenemos en sale order type.
Deprecamos directamente el feature y movemos algunas cosas a l10n_ar_sale_order_type

jjscarafia commented:
mezclamos igual, el error es por borrado de módulo
